Secret-agency-persistence/src/main/cz/fi/muni/pa165/secretagency/entity/Mission.java

```diff
@@ -0,0 +1,129 @@
+package main.cz.fi.muni.pa165.secretagency.entity;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+/**
+ * Mission entity for Secret agency project.
+ *
+ * @author Adam Skurla (487588)
+ */
+public class Mission {
+    private Long id;
+    private Double[] location;
+    private String missionType;
+    private LocalDate started;
+    private LocalDate ended;
+
+    /**
+     * Constructor
+     * @param id of mission
+     * @param location of mission (latitude and longtitude)
+     * @param missionType type of mission
+     * @param started mission start date
+     * @param ended mission end date
+     */
+    public Mission(Long id, Double[] location, String missionType, LocalDate started, LocalDate ended) {
+        this.id = id;
+        this.location = location;
+        this.missionType = missionType;
+        this.started = started;
+        this.ended = ended;
+    }
+
+    /**
+     * Create empty Mission
+     */
+    public Mission() {
+    }
+
+    /**
+     * @return mission id
+     */
+    public Long getId() {
+        return id;
+    }
+
+    /**
+     * @param id mission id
+     */
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    /**
+     * @return mission location
+     */
+    public Double[] getLocation() {
+        return location;
+    }
+
+    /**
+     * @param location of mission
+     */
+    public void setLocation(Double[] location) {
+        this.location = location;
+    }
+
+    /**
+     * @return mission type
+     */
+    public String getMissionType() {
+        return missionType;
+    }
+
+    /**
+     * @param missionType type of mission
+     */
+    public void setMissionType(String missionType) {
+        this.missionType = missionType;
+    }
+
+    /**
+     * @return mission start date
+     */
+    public LocalDate getStarted() {
+        return started;
+    }
+
+    /**
+     * @param started start date of mission
+     */
+    public void setStarted(LocalDate started) {
+        this.started = started;
+    }
+
+    /**
+     * @return mission end date
+     */
+    public LocalDate getEnded() {
+        return ended;
+    }
+
+    /**
+     * @param ended end date of mission
+     */
+    public void setEnded(LocalDate ended) {
+        this.ended = ended;
+    }
+
+    @Override
```

Comment:
Na cvičení nebo v přednášce říkali, že by se mělo k třídním atributům přistupovat přes gettery. Důvod byl myslím takový, že Hibernate může vytvářet proxy. Ze stejného důvodu taky používat instanceOf, místo porovnávání přes getClass().

Reply:
Promiň za špatný označení, jde mi o equals.

Reply:
Btw. Idea umí takový equals a hashcode vygenerovat, jen tam musíš zaškrtnou checkboxy.

Reply:
možeš prosim označiť riadok, ktoreho sa to týka? Takto neviem uplne čo mám spraviť..

Reply:
tak som to opravil, som si to az potom vsimol moja chyba :)